geth/api/backend.go

```diff
@@ -251,16 +253,32 @@ func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedEx
 	return selectedAccount, nil
 }
 
-// ApproveSignRequest instructs backend to complete sending of a given transaction
-func (b *StatusBackend) ApproveSignRequest(id string, password string) sign.Result {
-	return b.pendingSignRequests.Approve(id, password, b.getVerifiedAccount)
+// prepareTxArgs given gas and gasPrice will prepare a valid sign.TxArgs.
+func (b *StatusBackend) prepareTxArgs(gas, gasPrice int64) (args sign.TxArgs) {
+	if gas > 0 {
+		g := hexutil.Uint64(gas)
+		args.Gas = &g
+	}
+	if gasPrice > 0 {
+		gp := (*hexutil.Big)(big.NewInt(gasPrice))
+		args.GasPrice = gp
+	}
+	return
+}
+
+// ApproveSignRequest instructs backend to complete sending of a given transaction.
+// Empty values for gas or gasPrice will preserve values for these properties
+// as they were defined on the transaction initialization.
+func (b *StatusBackend) ApproveSignRequest(id, password string, gas, gasPrice int64) sign.Result {
```

Comment:
Or we could create a new method `ApproveSignRequestWithArgs()` where args would be defined as a struct. This way, we could leave this method unchanged and avoid things like `results[txID] = b.ApproveSignRequest(txID, password, 0, 0)`.